scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -766,6 +770,211 @@ public DatasetId storeSimulationResults(final PlanMetadata planMetadata,
     return datasetIds.datasetId();
   }
 
+  @Override
+  public Optional<SimulationResults> getSimulationResults(PlanMetadata planMetadata)
+  throws PlanServiceException, IOException, InvalidJsonException
+  {
+    final var simulationDatasetId = getSuitableSimulationResults(planMetadata);
+    if(simulationDatasetId.isEmpty()) return Optional.empty();
+    final var request = """
+        {
+          simulation(where: {plan_id: {_eq: %d}}) {
+            id
+            simulation_datasets(where: {plan_revision: {_eq: %d}, canceled: {_eq: false}, dataset_id: {_eq: %d}}) {
+              id
+              simulation_start_time
+              simulation_end_time
+              simulated_activities {
+                activity_directive {
+                  id
+                  arguments
+                  type
+                  anchored_to_start
+                  anchor_id
+                }
+                activity_type_name
+                duration
+                id
+                parent_id
+                start_offset
+                attributes
+              }
+              dataset {
+                profiles {
+                  type
+                  profile_segments {
+                    start_offset
+                    dynamics
+                  }
+                  name
+                }
+                spans(where: {duration: {_is_null: true}}) {
+                attributes
+                parent_id
+                type
+                start_offset
+                id
+              }
+              }
+
+            }
+          }
+        }""".formatted(planMetadata.planId().id(), planMetadata.planRev(), simulationDatasetId.get().id());
+
+    final JsonObject response;
+    response = postRequest(request).get();
+    final var data = response.getJsonObject("data");
```

Comment:
Reminder to split this into several requests (optionally made and parsed in parallel) to significantly improve performance.